src/haar_filter.jl

```diff
@@ -26,12 +26,14 @@ RadiationDetectorDSP.flt_output_time_axis(fi::HaarTypeWaveletFilterInstance, tim
 function RadiationDetectorDSP.rdfilt!(output::AbstractVector{T}, fi::HaarTypeWaveletFilterInstance{T}, input::AbstractVector{T}) where T
     invsqrt = inv(sqrt(T(fi.length)))
     @assert flt_output_length(fi) == length(output) "output length not compatible with filter"
-    Δidx = fi.length - 1
-    output[1] = (input[1] + input[1+Δidx]) * invsqrt
-    output[end] = (input[end] + input[end-Δidx]) * invsqrt
-    for i in 2:length(output)-1
-        j = (i-1)*fi.down_sampling_rate + 1
-        output[i] = sum(view(input, j:j+Δidx)) * invsqrt
+    @inbounds @simd for i in eachindex(output)
```

Comment:
We should check `axes(input) == axes(output)` explicitly before though, otherwise using `@inbounds` here is very dangerous.

Reply:
`axes(input)` is not equal `axes(output)`, that's the whole point of the filter (sampling it down).
The `@inbounds` should not be a problem because:
- `i` (used to access `output`) is iterating through `eachindex(output)`
- `j` (used to access `input`) is forced to be between `1` and `end` of `input` using the `min(end, max(1, j))` syntax.

Reply:
We could check `@assert length(input) == length(output) * fi.down_sampling_rate` explicitly.

@esleon97: Is there any case in which `fi.down_sampling_rate` is different from (and especially smaller than) `fi.length`? The minimum index to access `input` is `1`, the maximum index to access `input` is `(length(output) - 1)*fi.down_sampling_rate + fi.length`.
If `fi.length` is always equal or smaller than `fi.down_sampling_rate`, then we will never access `input` outside of its bounds and we could also get rid of the `min(end, max(...)` syntax.

Reply:
No, I think that `fi.length` will always be equal to `fi.down_sampling_rate`.

Reply:
> axes(input) is not equal axes(output), that's the whole point of the filter 

Ah, ok. But if the just stores the scaling coefficients, not the wavelet coefficients, then we should find a more fitting name than `HaarTypeWaveletFilter`. We can just call it a summing filter or averaging filter.

Reply:
I'll let that up to you to decide. Feel free to push changes to this PR, my part here is done ;)

Reply:
@fhagemann thanks! @oschulz maybe we can name it `HaarAveragingFilter` ?